common/include/pcl/common/impl/io.hpp

```diff
@@ -118,6 +118,25 @@ getFieldsList (const pcl::PointCloud<PointT> &)
 }
 
 
+template <typename PointInT, typename PointOutT> static void
+copyPointCloudMemcpy (const pcl::PointCloud<PointInT> &cloud_in,
+                      pcl::PointCloud<PointOutT> &cloud_out)
+{
+  // Iterate over each point, if the point types of two clouds are different
+  for (std::size_t i = 0; i < cloud_in.size (); ++i)
+    copyPoint (cloud_in[i], cloud_out[i]);
+}
+
+
+template <typename PointT> static void
+copyPointCloudMemcpy (const pcl::PointCloud<PointT> &cloud_in,
+                      pcl::PointCloud<PointT> &cloud_out)
+{
+  // Use std::copy directly, if the point types of two clouds are same
+  std::copy (&cloud_in[0], (&cloud_in[0]) + cloud_in.size (), &cloud_out[0]);
+}
```

Comment:
Should we move them into a detail namespace? (Technically it's in hpp file, so it should be ok to leave it as it is). Thoughts @mvieth ?

Reply:
Good idea :+1:

Reply:
Thanks for suggesting to move the functions into detail namespace, which really makes the code nicer! :grinning: I've done that.